wagtail/snippets/tests/test_viewset.py

```diff
@@ -22,20 +22,27 @@
 from wagtail.test.utils import WagtailTestUtils
 
 
-class TestCustomIcon(WagtailTestUtils, TestCase):
+class BaseSnippetViewSetTests(WagtailTestUtils, TestCase):
```

Comment:
Not sure if this encompasses the search tests that now have to be done as a TransactionTestCase instead as of #10208 - will find out when I rebase... - but if it does, I think the work needed to untangle that might outweigh the benefit of pulling out these methods, in which case I might skip this commit.